contracts/colonyNetwork/ColonyNetworkExtensions.sol

```diff
@@ -0,0 +1,135 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "../colony/ColonyDataTypes.sol";
+import "../colonyNetwork/IColonyNetwork.sol";
+import "../extensions/ColonyExtension.sol";
+import "./ColonyNetworkStorage.sol";
+
+
+contract ColonyNetworkExtensions is ColonyNetworkStorage {
+
+  // Public functions
+
+  function addExtension(bytes32 _extensionId, address _resolver, bytes32 _roles)
+    public
+    stoppable
+    calledByMetaColony
+  {
+    require(_resolver != address(0x0), "extension-manager-bad-resolver");
+
+    uint256 version = getResolverVersion(_resolver);
+    require(version == 1 || _roles == 0, "extension-manager-nonempty-roles");
+    require(version == 1 || resolvers[_extensionId][version - 1] != address(0x0), "extension-manager-bad-version");
```

Comment:
I would actually leave out that check, since that would preclude updating resolvers, which we allow for Colony versions:

```
  function addColonyVersion(uint _version, address _resolver) public
  always
  calledByMetaColony
  {
    require(currentColonyVersion > 0, "colony-network-not-intialised-cannot-add-colony-version");

    colonyVersionResolver[_version] = _resolver;
    if (_version > currentColonyVersion) {
      currentColonyVersion = _version;
    }

    emit ColonyVersionAdded(_version, _resolver);
  }
```

Thinking more, perhaps even requiring resolvers to be added in sequence is unecessary.

Reply:
Further, storing the roles is also not really necessary since they're never used for anything consensus-relevant, it's really just documentation.

Reply:
> Thinking more, perhaps even requiring resolvers to be added in sequence is unecessary.

Surely it definitely is if we only allow upgrading one-at-a-time?

I think allowing overwriting of a version's resolving is asking for a whole load of trouble to be honest, and it probably shouldn't be allowed on the colony version side either.

Reply:
I would agree, but while the logic for colony resolvers is as permissive as it is, I don't think it makes sense to have the logic for extensions be more restrictive. My preference would be to keep them at the same level of security and update them both in parallel once we're ready to do so.